### ISSUE TITLE
Enforcing reviews for stories

### ISSUE THREAD

**Issue by ttplaya** [MicrosoftDocs/vsts-docs#3028]
I'm trying to setup rules in TFS2018 to make sure that every story that gets created gets reviewed by a team lead before it gets worked on. I've added a Reviewed state and set the following state rule for the New state:

            &lt;FIELD refname="System.State"&gt;
              &lt;READONLY not="[team leads group]" /&gt;
            &lt;/FIELD&gt;

So this is supposed to restrict anyone besides the team leads from moving the story out of New state. However, I find that a user can still go to the agile board view and drag the story to another column, and the state would still get changed. Is there a better way to do what I'm trying to do?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c8346f1b-7276-2508-eab9-85d695b21a8e
* Version Independent ID: 9c5a5f6f-5c67-26e7-f00b-74d10a207310
* Content: [Apply a field rule - Azure DevOps & TFS](https://docs.microsoft.com/en-us/azure/devops/reference/xml/apply-rule-work-item-field?toc=%2Fazure%2Fdevops%2Freference%2Ftoc.json&bc=%2Fazure%2Fdevops%2Freference%2Fbreadcrumb%2Ftoc.json&view=tfs-2018#scope)
* Content Source: [docs/reference/xml/apply-rule-work-item-field.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/reference/xml/apply-rule-work-item-field.md)
* Product: **devops**
* GitHub Login: @KathrynEE
* Microsoft Alias: **kaelli**

**Comment by ttplaya** [comment]
I think I've found the way to do what I want to do:

1. Add a new field called ApprovedBy, that has the following rules in the field definition:
        <ALLOWEXISTINGVALUE />
        <ALLOWEDVALUES expanditems="true" filteritems="excludegroups">
          <LISTITEM value="[team leads group]" />
        </ALLOWEDVALUES>
        <VALIDUSER />
        <WHENNOT field="System.State" value="New">
          <READONLY not="[team leads group]" />
        </WHENNOT>

2. Set the following state rule for the New state:
            <FIELD refname="ApprovedBy">
              <EMPTY />
            </FIELD>

3. Set the following state rule for the rest of the states:
            <FIELD refname="ApprovedBy">
              <ALLOWEDVALUES expanditems="true" filteritems="excludegroups">
                <LISTITEM value="[team leads group]" />
              </ALLOWEDVALUES>
            </FIELD>

4. For the Reviewed, Active, and Resolved states, add the <REQUIRED /> rule to the above for the ApprovedBy field

5. For the transitions from New to the Reviewed, Active, and Resolved states, add the following rule:
            <FIELD refname="ApprovedBy">
              <COPY for="[team leads group]" from="currentuser" />
            </FIELD>
 
Now, whenever anyone tries to move a story out of New state into Reviewed, Active, or Resolved, they will not be able to do so because the ApprovedBy field is required but is readonly for them. Only the team leads are able to move the story out of New into those states.

Note that what I've mentioned is only a partial solution -- the part that deals with preventing non-privileged users from approving stories and automatically populating the ApprovedBy field. I also had to tweak a bunch of other stuff in the transition rules to make sure the transitions made sense with the addition of the new Reviewed state.
